WWST-1263: ZW121-A Aeotec LED Strip - US

joshk-trility opened the pull request:
Updated the DTH for the Aeon LED Bulb to support the LED Strip.
Added multi-attribute tiles on details screen. Fixed some logging,
cleaned up unused tiles.
Also fixed occasional error when setting the color (WWST-1418).

joshk-trility committed:
WWST-1263: ZW121-A Aeotec LED Strip - US

Updated the DTH for the Aeon LED Bulb to support the LED Strip.
Added multi-attribute tiles on details screen. Fixed some logging,
cleaned up unused tiles.
Also fixed WWST-1418.

jrrall committed:
Added crc encapsulation

jrrall committed:
Cleanup, removed commenting out of led-strip fingerprints, and Gen6 from name

jrrall committed:
reverted some stuff

jrrall committed:
reverted some stuff

jrrall committed:
Removed secstate hack

jrrall committed:
Removed secstate hack

joshk-trility committed:
Merge pull request #1 from jrrall/WSST-1277-2

WSST-1277 CRC Encap

joshk-trility committed:
Fix indentation and check interval value.

joshk-trility committed:
Fixed color temperature calculations and controls.

Updated installed state to reflect correct initial values.
Fixed various other minor issues (indentation, etc).

greens commented on devicetypes/smartthings/aeon-led-bulb.src/aeon-led-bulb.groovy:
```diff
@@ -17,67 +17,78 @@
-		state "off", label:'${name}', action:"switch.on", icon:"st.lights.philips.hue-single", backgroundColor:"#ffffff", nextState:"turningOn"
-		state "turningOn", label:'${name}', action:"switch.off", icon:"st.lights.philips.hue-single", backgroundColor:"#00a0dc", nextState:"turningOff"
-		state "turningOff", label:'${name}', action:"switch.on", icon:"st.lights.philips.hue-single", backgroundColor:"#ffffff", nextState:"turningOn"
+    tiles(scale: 2) {
```
fix spaces=>tabs

greens replied:
happens a couple of other places

joshk-trility replied:
Done.

greens commented on devicetypes/smartthings/aeon-led-bulb.src/aeon-led-bulb.groovy:
```diff
@@ -182,30 +239,51 @@ def setColor(value) {
-	if(value.hex) sendEvent(name: "color", value: value.hex)
-	if(value.switch) sendEvent(name: "switch", value: value.switch)
-	if(value.saturation) sendEvent(name: "saturation", value: value.saturation)
+	result += queryAllColors()
```
does this need a delay, in your experience?

joshk-trility replied:
The commands() call right after it will add a 200ms delay. queryAllColors() just builds a list of switchColorGet commands.

greens replied:
👍

joshk-trility committed:
Fix indentation and refresh event interval.

The time between commands was 1 second, which would lead to the
refresh request taking over 5 seconds. Removed the parameter to
use the default of 200ms.

joshk-trility reviewed (commented):
New commit incoming with minor changes.

greens commented on devicetypes/smartthings/aeon-led-bulb.src/aeon-led-bulb.groovy:
```diff
@@ -17,67 +17,78 @@
+def installed() {
+	log.debug "installed()..."
+	state.colorReceived = ["red": null, "green": null, "blue": null, "warmWhite": null, "coldWhite": null]
+	sendEvent(name: "checkInterval", value: 1860, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
```
set offlinePingable to 0. We're mostly using this for health/safety-critical devices.

joshk-trility replied:
Is "offlinePingable" only used for FLiRS devices? And, is leaving it out of that map the same as setting it to "0"?

greens replied:
We're using it to indicate to Device Health which devices it should ping when the hub returns online after being offline. This is supposed to be for devices which may have life safety implications (like locks) where missing a state change in such an event would be bad.

greens commented on devicetypes/smartthings/aeon-led-bulb.src/aeon-led-bulb.groovy:
```diff
@@ -182,30 +239,51 @@ def setColor(value) {
 
 private command(physicalgraph.zwave.Command cmd) {
-	if (state.sec != 0) {
-		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
```
So this is the one place we have to worry about backward compatibility. Any device that was added before we started distinguishing secure devices via the `zwaveInfo` won't pass the below test, so we still need to have the check for `state.sec`.

joshk-trility commented:
Due to major differences between the LED bulb and LED strip, I'm closing this pull request and resubmitting the LED strip changes under the generic RGBW light DTH.
